### DISCUSSION TITLE
How to Contribute to the Pattern library

### DISCUSSION THREAD

**Discussion by ishah-twilio** [twilio-labs/paste#1088] [Office hours] [answered]
Hello there,  

Mindy and Ishita like to contribute to Edit Object pattern. 

Our upcoming projects: 
Mindy — Email Product Design | Event Streams
Ishita — IoT > Super SIM > Fleets

Time each of us can spend:
Mindy — probably 1-2 hours/week
Ishita — 4 hours / week  

We like DSys help and guidance to understand the scope to contribute “Edit Object” pattern to the pattern library.

**Comment by serifluous** [comment, marked as answer]
This is great, especially since this'll be a cross-product effort. Thanks for reaching out!

1. You can start by reaching out to product designers across Console, SendGrid, and Flex for examples of "Edit Object" use cases—The screenshots in the airtable link I shared in Slack can be a good starting point.
2. After that, draft a spec ([template here](https://docs.google.com/document/d/1SSIURVMzlGbANyvtYlU61ZdDaSqF2QigFdFZuSnYSkk/edit?usp=sharing)) and mockups of the pattern, which we can look over together at office hours.
3. An accessibility + systemizability review is all we need to do to publish an "alpha" pattern. However even for alpha patterns, we like to be reasonably confident that most PDs agree with the pattern. If after asking around or showing the spec and design at a few crits (or even if you have a hunch), you feel like there's not going to be strong alignment among the PDs, consider what you might publish as a basic MLP/MVP version of the spec to start. Alpha patterns can always be iterated on and revised with additional research! Let us know if you get to this point and need some support.

If you're able to make a lengthier time commitment, you can also try getting the pattern to "beta". More details about how to do that starting here: https://paste.twilio.design/patterns/contribute#testing-and-collecting-feedback

Feel free to use this discussion thread as a work log if you want to share what you found after asking around and your draft spec as you go.
